packages/devtools_app/lib/src/shared/diagnostics/diagnostics_node.dart

```diff
@@ -589,9 +589,12 @@ class RemoteDiagnosticsNode extends DiagnosticableTree {
       inHideableGroup && !isHideableGroupLeader && groupIsHidden;
 
   bool get inHideableGroup {
-    final hasAtMostOneChild = childrenNow.length <= 1;
-    final isOnlyChild = (parent?.childrenNow ?? []).length == 1;
-    return !isCreatedByLocalProject && hasAtMostOneChild && isOnlyChild;
+    if (_alwaysVisible(this)) return false;
+    final parentIsHideable = parent != null && !_alwaysVisible(parent!);
+    final childIsHideable =
+        childrenNow.isNotEmpty && !_alwaysVisible(childrenNow.first);
+
+    return hasChildren ? parentIsHideable : parentIsHideable || childIsHideable;
```

Comment:
will `childIsHideable` ever be true if `hasChildren` is false? Since `childIsHideable` requires that `childrenNow.isNotEmpty`, I would think that this actually equates to:

`return hasChildren ? parentIsHideable : parentIsHideable || false`, which could just be `return parentIsHideable`.

Reply:
Haha yes this is exactly the logic that was wrong. I noticed the tree was showing some widgets it wasn't supposed to, and then realized I had written the ternary backwards 🤦‍♀️ 

It is now:

`hasChildren ? parentIsHideable || firstChildIsHideable : parentIsHideable`